src/watertap_contrib/seto/unit_models/surrogate/tests/test_med_tvc_surrogate.py

```diff
@@ -0,0 +1,342 @@
+        )
+        with pytest.raises(ValueError, match=error_msg):
+            m.fs.med_tvc.number_effects.set_value(100)
+
+    @pytest.mark.unit
+    def test_build(self, MED_TVC_frame):
+        m = MED_TVC_frame
+
+        # test ports
+        port_lst = ["feed", "distillate", "brine", "steam", "motive"]
+        for port_str in port_lst:
+            port = getattr(m.fs.med_tvc, port_str)
+            assert isinstance(port, Port)
+            assert len(port.vars) == 3
+
+        # test statistics
+        assert number_variables(m) == 201
+        assert number_total_constraints(m) == 58
+        assert number_unused_variables(m) == 76  # vars from property package parameters
+
+    @pytest.mark.unit
+    def test_dof(self, MED_TVC_frame):
+        m = MED_TVC_frame
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.unit
+    def test_calculate_scaling(self, MED_TVC_frame):
+        m = MED_TVC_frame
+        calculate_scaling_factors(m)
+
+        # check that all variables have scaling factors
+        unscaled_var_list = list(unscaled_variables_generator(m))
+        assert len(unscaled_var_list) == 0
+
+        # check that all constraints have been scaled
+        unscaled_constraint_list = list(unscaled_constraints_generator(m))
+        assert len(unscaled_constraint_list) == 0
+
+    @pytest.mark.component
+    def test_var_scaling(self, MED_TVC_frame):
+        m = MED_TVC_frame
+        badly_scaled_var_lst = list(badly_scaled_var_generator(m))
+        assert badly_scaled_var_lst == []
+
+    @pytest.mark.component
+    def test_initialize(self, MED_TVC_frame):
+        m = MED_TVC_frame
+        initialization_tester(m, unit=m.fs.med_tvc, outlvl=idaeslog.DEBUG)
+
+    @pytest.mark.component
+    def test_solve(self, MED_TVC_frame):
+        m = MED_TVC_frame
+        results = solver.solve(m)
+
+        # Check for optimal solution
+        assert_optimal_termination(results)
+
+    @pytest.mark.component
+    def test_mass_balance(self, MED_TVC_frame):
+        m = MED_TVC_frame
+
+        med_tvc = m.fs.med_tvc
+
+        feed_flow_m3_hr = 277.777
+        dist_flow_m3_hr = 83.333
+        brine_flow_m3_hr = 194.444
+        cool_flow_m3_hr = 131.593
+
+        feed_mass_flow_tot = 78.93
+        cool_mass_flow_tot = 37.31
+        feed_mass_flow_tds = 2.70
+        brine_mass_flow_tds = 2.70
+        recovery = dist_flow_m3_hr / feed_flow_m3_hr
+
+        assert value(med_tvc.recovery_vol_phase[0, "Liq"]) == pytest.approx(
+            recovery, rel=1e-3
+        )
+        assert value(
+            pyunits.convert(
+                med_tvc.feed_props[0].flow_vol_phase["Liq"]
+                - med_tvc.distillate_props[0].flow_vol_phase["Liq"]
+                - med_tvc.brine_props[0].flow_vol_phase["Liq"],
+                to_units=pyunits.m**3 / pyunits.hr,
+            )
+        ) == pytest.approx(
+            feed_flow_m3_hr - dist_flow_m3_hr - brine_flow_m3_hr, rel=1e-3
+        )
```

Comment:
This works but you could also just check that mass in - mass out is approximately 0 (e.g., `==pytest.approx(1e-6, rel=1e-3)`)

Reply:
An additional test is added for it.